src/Fields/field.jl

```diff
@@ -139,13 +223,22 @@ architecture(o::OffsetArray) = architecture(o.parent)
 "Returns `f.data.parent` for `f::Field`."
 @inline Base.parent(f::Field) = f.data.parent
 
-"Returns a view over the interior points of the `field.data`."
-@inline interior(f::Field) = view(f.data, 1:f.grid.Nx, 1:f.grid.Ny, 1:f.grid.Nz)
+@inline interior_indices(loc, topo, N) = 1:N
+@inline interior_indices(::Type{Face}, ::Bounded, N) = 1:N+1
+
+@inline interior_parent_indices(loc, topo, N, H) = 1+H:N+H
+@inline interior_parent_indices(::Type{Face}, ::Bounded, N, H) = 1+H:N+1+H
 
-"Returns a reference to the interior points of `field.data.parent.`"
-@inline interiorparent(f::Field) = @inbounds f.data.parent[1+f.grid.Hx:f.grid.Nx+f.grid.Hx,
-                                                           1+f.grid.Hy:f.grid.Ny+f.grid.Hy,
-                                                           1+f.grid.Hz:f.grid.Nz+f.grid.Hz]
+"Returns a view of `f` that excludes halo points."
+@inline interior(f::Field{X, Y, Z}) where {X, Y, Z} = view(f.data, interior_indices(X, x_topology(f), f.grid.Nx), 
+                                                                   interior_indices(Y, y_topology(f), f.grid.Ny), 
+                                                                   interior_indices(Z, z_topology(f), f.grid.Nz))
+
+"Returns a reference (not a view) to the interior points of `field.data.parent.`"
+@inline interiorparent(f::Field{X, Y, Z}) where {X, Y, Z} = 
+    @inbounds f.data.parent[interior_parent_indices(X, x_topology(f), f.grid.Nx, f.grid.Hx),
+                            interior_parent_indices(Y, y_topology(f), f.grid.Ny, f.grid.Hy),
+                            interior_parent_indices(Z, z_topology(f), f.grid.Nz, f.grid.Hz)]
```

Comment:
Nice! Should we also export `interiorparent`?

Reply:
Perhaps we should from `Fields` (but not from top-level)

Reply:
Ah, we already export from `Fields.jl`. Can this comment be clarified?